api/src/main/java/com/voxelwind/api/game/level/block/data/Cake.java

```diff
@@ -1,20 +1,18 @@
 package com.voxelwind.api.game.level.block.data;
 
-import com.google.common.base.Preconditions;
-import com.voxelwind.api.game.level.block.BlockData;
+import com.voxelwind.api.game.Metadata;
 
 /**
  * Represents a block of cake.
  */
-public class Cake implements BlockData {
+public class Cake implements Metadata {
     private final int level;
 
     public static final Cake NEW = new Cake(0);
     public static final Cake ALMOST_EATEN = new Cake(6);
 
-    public static Cake ofStage(int data) {
-        Preconditions.checkArgument(data >= 0 && data < 7, "data is not valid (wanted 0-6)");
```

Comment:
The validation should be done in the generation part. The idea behind the of() static helpers is to keep them until they get replaced by a Builder. I simply missed the // TODO here